### PR TITLE
Implement DeleteItemAction plugin support

### PR DESCRIPTION
**Pull request by nrb**
Part of #2400 

This PR adds a new plugin type, the `DeleteItemAction`.

It does _not_ update the deletion process to run them, however. It only updates the plugin client management, frameworks, and data types to support the new plugin.

Leaving in draft until the design is approved.

### CODE REVIEW

File: pkg/plugin/clientmgmt/restartable_delete_item_action_test.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2018, 2019 the Velero contributors.
```
[the remark this answers is not in the document]
Comment (nrb) [reply pkg/plugin/clientmgmt/restartable_delete_item_action_test.go]: Done.

File: pkg/plugin/velero/delete_item_action.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package velero
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+
+	api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
```
Comment (ashish-amarnath) [inline pkg/plugin/velero/delete_item_action.go:22]: s/api/velerov1api

File: pkg/plugin/clientmgmt/manager.go [hunk 3]
```diff
@@ -224,3 +230,39 @@ func (m *manager) GetRestoreItemAction(name string) (velero.RestoreItemAction, e
...
+	return actions, nil
+}
+
+// GetDeleteItemAction returns a restartableDeleteItemAction for name.
+func (m *manager) GetDeleteItemAction(name string) (velero.DeleteItemAction, error) {
+	// Backwards compatibility with non-namespaced plugins, following principle of least surprise
+	// since DeleteItemActions were not bundled with Velero when plugins were non-namespaced.
+	if !strings.Contains(name, "/") {
+		name = "velero.io" + name
+	}
```
[the remark this answers is not in the document]
Comment (nrb) [reply pkg/plugin/clientmgmt/manager.go:260]: As I mentioned in the comment, I put this in to follow the principle of least surprise. Yes, it's a new plugin type, but having it deviate from the rest seems inconsistent and surprising should we go to add an in-tree plugin and it doesn't work.

I'll add the missing `/`, that was a typo.

File: pkg/plugin/clientmgmt/manager_test.go [hunk 4]
```diff
@@ -475,3 +475,85 @@ func TestGetRestoreItemActions(t *testing.T) {
 		})
 	}
 }
+
+func TestGetDeleteItemActions(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (nrb) [reply pkg/plugin/clientmgmt/manager_test.go:479]: That's not used on any of the other plural method tests in this file. I am, however, missing a test for `GetDeleteItemAction` I notice.

**Pull request merged**